paddle/phi/kernels/fusion/onednn/fused_matmul_kernel.cc

```diff
@@ -193,6 +207,28 @@ class FusedMatmulOneDNNHandler
         this->fwd_pd_->weights_desc(), funcs::to_void_cast<YT>(input_data));
   }
 
+  std::shared_ptr<dnnl::memory> AcquireSrcMemoryStride(
+      const DenseTensor *input) {
+    const XT *input_data = input->data<XT>();
+    std::shared_ptr<dnnl::memory> src_mem =
+        this->AcquireMemoryFromPrimitive(this->fwd_pd_->dst_desc());
+    auto residual_vec = vectorize(input->dims());
+    int IC = residual_vec[1];
+    int IH = residual_vec[2];
+    int IW = residual_vec[3];
+    size_t size = this->fwd_pd_->dst_desc().get_size() / sizeof(XT);
+    XT *dst = static_cast<XT *>(src_mem->get_data_handle());
+
+#pragma omp parallel for
+    for (size_t i = 0; i < size; ++i) {
+      auto mod_i =
+          static_cast<int>(i - floor(i / (IC * IH * IW)) * (IC * IH * IW));
+      // Use stride to make 1*C*H*W to N*C*H*W to avoid broadcast overhead
+      dst[i] = input_data[mod_i];
+    }
```

Comment:
seems here changes the data handle instead of the stride?

Reply:
Do you mean directly change the stride of mem? That is also a good method, but I am not sure if it's intuitive? This kind of change in PR can also attain a big improvement, though yours seems bring less overhead...

Reply:
I think both are okay, but the code change is not align with the comments. Maybe adjust the comments will be better. BTW, is openmp always available in Paddle?

Reply:
Seems they diasble openmp in WIN32 case:
<img width="304" alt="image" src="https://github.com/PaddlePaddle/Paddle/assets/115513981/7ade9e24-d45b-4ba2-bfbe-d4555f30a8fc">
Maybe adding
`#if defined(_OPENMP)`
could be better.